### PR TITLE
seperate docker installs from run

### PR DESCRIPTION
**Pull request by andykais**
take npm installs and pip installs out of `docker-compose.yml` and into `docker-init.yml`

to run okcollege-dev now:
on INIT `./bin/okcollege-init` (`docker-compose -f docker-init.yml`)
on each run `./bin/okcollege` (`docker-compose up`)

### CODE REVIEW

**Review by seveibar — CHANGES_REQUESTED** [review]
after docker-prod will accept

**Review by seveibar — APPROVED** [review]

File: bin/okc-psql [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-docker run -it --rm --net host postgres psql -h localhost -U postgres
+docker run -it --rm --net host postgres psql --command "CREATE DATABASE okcollege_dev;" -h localhost -U postgres
```
Comment (seveibar) [inline bin/okc-psql]: this file is for debugging the database, not creating a table in the database.

better with a separate bin file for the database creation or embed it directly in the docker-compose

File: bin/okcollege-init [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+cd "$(dirname "$0")"
+cd ..
+docker-compose -f docker-init.yml up
```
Comment (seveibar) [inline bin/okcollege-init:5]: yes

File: bin/okcollege-prod [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+cd "$(dirname "$0")"
+cd ..
+docker-compose -f ./docker-prod.yml up
```
Comment (seveibar) [inline bin/okcollege-prod:5]: yes

File: docker-compose.yml [hunk 4]
```diff
@@ -8,7 +8,7 @@ services:
     volumes:
       - ./web-client:/usr/src/app
       - ./assets:/usr/src/app/assets
-    command: "/bin/bash -c \"cd /usr/src/app && npm install && npm run build:dll && npm start\""
+    command: "/bin/bash -c \"cd /usr/src/app && npm start\""
```
Comment (seveibar) [inline docker-compose.yml:11]: yes

File: docker-init.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    env_file: development.env
+  predictor-preprocessor:
+    image: python:3.5
+    network_mode: host
+    environment:
+      PYTHONUSERBASE: /usr/src/app/vendor
+    volumes:
+      - ./assets:/usr/src/app/assets
+      - ./predictor-preprocessor:/usr/src/app
+    command: "/bin/bash -c \"cd /usr/src/app && pip install --upgrade pip && pip install -r requirements.txt --user\""
```
Comment (seveibar) [inline docker-init.yml:58]: yes

**Review by seveibar — CHANGES_REQUESTED** [review]

**Comment by seveibar** [comment]
~~I can't say I like the output being hidden, especially for `docker-init.yml`, I put the web-client on my system in a bad state because I wasn't sure if the init script was working or not and stopped it half way through (easily resolved, but emphasizes the problem).~~

This turned out to be a system specific problem

**Comment by seveibar** [comment]
This has been tested on my system and works well :+1:

**Comment by andykais** [comment]
`okc-psql` and `okc-psql-init` files are separated, is that all you wanted?

**Review by seveibar — APPROVED** [review]

**Pull request merged**